src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -1979,6 +1979,44 @@ public void actionPerformed(final ActionEvent e) {
     }
   };
 
+  /**
+   * Prompt user for which producer territory (factory) to use when producing units in sea
+   * zones that have multiple adjacent factories.
+   * 
+   * @param candidates
+   *        - list of producer territories to choose from
+   * @param unitTerritory
+   *        - territory units are being placed
+   * @return territory selected to use for production
+   */
+  public Territory selectProducerTerritoryForUnits(final Collection<Territory> candidates,
```

Comment:
Here is the method that actually puts of the popup to the user with a list of territories to select from.